runtime/parachains/src/ump.rs

```diff
@@ -408,22 +404,29 @@ impl<T: Config> Pallet<T> {
 			};
 
 			// dequeue the next message from the queue of the dispatchee
-			let (upward_message, became_empty) = queue_cache.dequeue::<T>(dispatchee);
-			if let Some(upward_message) = upward_message {
+			let maybe_next = queue_cache.peek_front::<T>(dispatchee);
+			let became_empty = if let Some(upward_message) = maybe_next {
 				match T::UmpSink::process_upward_message(
 					dispatchee,
 					&upward_message[..],
```

Comment:
Now `upward_message` is already a reference we can simplify to:
```suggestion
					upward_message,
```